pkg/util/goroutinemap/goroutinemap_test.go

```diff
@@ -106,9 +153,63 @@ func Test_NewGoRoutineMap_Positive_SecondOpAfterFirstPanics(t *testing.T) {
 	}
 }
 
+func Test_NewGoRoutineMap_Positive_SecondOpAfterFirstPanicsWithExpBackoff(t *testing.T) {
+	// Arrange
+	grm := NewGoRoutineMap(true /* exponentialBackOffOnError */)
+	operationName := "operation-name"
+	operation1 := generatePanicFunc()
+	err1 := grm.Run(operationName, operation1)
+	if err1 != nil {
+		t.Fatalf("NewGoRoutine failed. Expected: <no error> Actual: <%v>", err1)
+	}
+	operation2 := generateNoopFunc()
+
+	// Act
+	err2 := retryWithExponentialBackOff(
+		time.Duration(500*time.Millisecond), // Longer duration to accommodate for backoff
```

Comment:
Can we put 500 as constant at the beginning? Also should we use a number larger than 500 to make sure it passed the retry period?